pkg/binary/envoy/debug/lsof.go

```diff
@@ -83,26 +73,25 @@ func (o *openFilesDataCollection) retrieveOpenFilesData() error {
 	defer cancel()
 
 	envoyProcess, err := process.NewProcessWithContext(ctx, pid)
-	if logDebugOnError(ctx, err, "process", pid) {
-		return nil
+	if w := wrapError(ctx, err, "process", pid); w != nil {
+		return w
 	}
 
 	result := make([]OpenFileStat, 0)
-	// print open file stats for all envoy instances
-	// relevant fields of the process
+	// print open file stats for all envoy relevant fields of the process
 	username, err := envoyProcess.UsernameWithContext(ctx)
-	if logDebugOnError(ctx, err, "username", pid) {
-		return nil
+	if w := wrapError(ctx, err, "username", pid); w != nil {
```

Comment:
wrapError is defined in node.go but used both in lsof.go and node.go which seems weird to me..

Reply:
I can extract them into a different file

Reply:
sg